### ISSUE TITLE
[Bug] Per-layer schedulers are built with doubled warmup and training steps but only advance once per update

### ISSUE THREAD

**Issue by gss10282023** [zhuhanqing/APOLLO#14]
### Summary

In `utils/setup.py`, the `q_apollo_per_layer` and `galore_adamw8bit_per_layer` branches build each parameter's scheduler with

```python
num_training_steps=args.num_training_steps * 2,
warmup_steps=args.warmup_steps * 2,
```

In the supported `gradient_accumulation=1` configuration each of these schedulers advances exactly once per optimizer update, so the learning-rate schedule runs at half the configured pace. The `q_galore_adamw8bit_per_layer` branch carries the same multiplier; I did not exercise that branch here.

### Environment

Commit [830f7646](https://github.com/zhuhanqing/APOLLO/blob/830f7646472497c742ffd5863e18e5156a46a5cc/utils/setup.py), PyTorch 2.3.0+cu121, bitsandbytes 0.43.1, Transformers 4.45.2, RTX A6000, FP32.

### Steps to reproduce

Use either affected optimizer with `gradient_accumulation=1`, learning rate `0.005`, `warmup_steps=2`, `num_training_steps=12`. Run two updates and read a per-parameter scheduler after each `scheduler.step()`:

| Scheduler steps taken | Current LR | LR from the configured schedule |
| --- | ---: | ---: |
| 1 | `0.00125` | `0.0025` |
| 2 | `0.0025` | `0.005` |

Optimizer and scheduler step counters both read 2, so there is no second scheduler call that the doubling could be compensating for.

In a full run of the training entry with a two-layer LLaMA (hidden size 128, vocabulary 512), the parameter updates differ from the configured-schedule reference by `0.5001851` relative L2 for `q_apollo_per_layer` and `0.499999997` for `galore_adamw8bit_per_layer`.

### Expected behavior

Schedulers should be built with the declared `warmup_steps` and `num_training_steps`.

### Proposed fix

Remove the two `* 2` factors in each affected scheduler constructor. Nothing else changes: gradients, optimizer math and projection-update intervals stay as they are.

After the change, the model, optimizer and scheduler states match the configured-schedule reference exactly for both branches, and a constant-LR control is unaffected. Reproduced on a second host. This is a small-model check; I have not tried to attribute any published result to this setting.
